[receiver/rabbitmqreceiver] Integration test fails on TestRabbitmqIntegration

### What happened?

## Description
Integration test fails on TestRabbitmqIntegration

## Steps to Reproduce

## Expected Result
Passed the test
## Actual Result
=== RUN   TestRabbitmqIntegration
=== RUN   TestRabbitmqIntegration/Running_rabbitmq_3.9
=== PAUSE TestRabbitmqIntegration/Running_rabbitmq_3.9
=== CONT  TestRabbitmqIntegration/Running_rabbitmq_3.9
2022/10/04 16:40:10 Starting container id: 73a35bb4430c image: docker.io/testcontainers/ryuk:0.3.3
2022/10/04 16:40:10 Waiting for container id 73a35bb4430c image: docker.io/testcontainers/ryuk:0.3.3
2022/10/04 16:40:11 Container is ready id: 73a35bb4430c image: docker.io/testcontainers/ryuk:0.3.3
2022/10/04 16:40:18 Starting container id: 7bfe81c274e1 image: d4163118-50c4-4d66-bb89-cc94b8419130:d4ebb5fc-5cf9-4871-845f-fc9e0f79b31b
2022/10/04 16:40:18 Waiting for container id 7bfe81c274e1 image: d4163118-50c4-4d66-bb89-cc94b8419130:d4ebb5fc-5cf9-4871-845f-fc9e0f79b31b
2022/10/04 16:40:21 Container is ready id: 7bfe81c274e1 image: d4163118-50c4-4d66-bb89-cc94b8419130:d4ebb5fc-5cf9-4871-845f-fc9e0f79b31b
    integration_test.go:99: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/rabbitmqreceiver/integration_test.go:99
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/rabbitmqreceiver/integration_test.go:53
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 64
        	Test:       	TestRabbitmqIntegration/Running_rabbitmq_3.9
--- FAIL: TestRabbitmqIntegration (0.00s)
    --- FAIL: TestRabbitmqIntegration/Running_rabbitmq_3.9 (12.20s)

### Collector version

v0.61.0

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

workflow:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3183783304/jobs/5191441744
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3183620109/jobs/5191141258

--- a/receiver/rabbitmqreceiver/integration_test.go
+++ b/receiver/rabbitmqreceiver/integration_test.go
@@ -19,10 +19,14 @@ package rabbitmqreceiver
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"io"
 	"path/filepath"
+	"strings"
 	"testing"
 	"time"
+	"unicode"
 
 	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go"
@@ -42,8 +46,7 @@ var (
 		},
 		ExposedPorts: []string{"15672:15672"},
 		Hostname:     "localhost",
-		WaitingFor: wait.ForListeningPort("15672").
-			WithStartupTimeout(2 * time.Minute),
+		WaitingFor:   waitStrategy{},
 	}
 )
 
@@ -54,6 +57,8 @@ func TestRabbitmqIntegration(t *testing.T) {
 		defer func() {
 			require.NoError(t, container.Terminate(context.Background()))
 		}()
+		require.NoError(t, container.Start(context.Background()))
+
 		hostname, err := container.Host(context.Background())
 		require.NoError(t, err)
 
@@ -93,12 +98,36 @@ func getContainer(t *testing.T, req testcontainers.ContainerRequest) testcontain
 			Started:          true,
 		})
 	require.NoError(t, err)
-
-	code, _, err := container.Exec(context.Background(), []string{"./setup.sh"})
-	require.NoError(t, err)
-	require.Equal(t, 0, code)
-
-	err = container.Start(context.Background())
-	require.NoError(t, err)
 	return container
+}
+
+type waitStrategy struct{}
+
+func (ws waitStrategy) WaitUntilReady(ctx context.Context, st wait.StrategyTarget) error {
+	if err := wait.ForListeningPort("15672").
+		WithStartupTimeout(2*time.Minute).
+		WaitUntilReady(ctx, st); err != nil {
+		return err
+	}
+
+	code, r, err := st.Exec(context.Background(), []string{"./setup.sh"})
+	if err != nil {
+		return err
+	}
+	if code == 0 {
+		return nil
+	}
+
+	// Try to read the error message for the sake of debugging
+	if errBytes, readerErr := io.ReadAll(r); readerErr == nil {
+		// Error message may have non-printable chars, so clean it up
+		errStr := strings.Map(func(r rune) rune {
+			if unicode.IsPrint(r) {
+				return r
+			}
+			return -1
+		}, string(errBytes))
+		return errors.New(strings.TrimSpace(errStr))
+	}
+	return errors.New("setup script returned non-zero exit code")
 }

--- a/receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh
+++ b/receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh
@@ -5,12 +5,18 @@ set -e
 USER='otelu'
 ROOT_PASS='otelp'
 
-echo "creating user: ${USER} . . ."
-rabbitmqctl add_user ${USER} ${ROOT_PASS} > /dev/null
+# Test if rabbitmqctl is working at all
+rabbitmqctl list_users > /dev/null
 
-echo "Configuring ${USER} permissions. . ."
-rabbitmqctl set_user_tags ${USER} administrator > /dev/null
-rabbitmqctl set_permissions -p /  ${USER} ".*" ".*" ".*" > /dev/null
+# Don't recreate user if already exists
+if ! rabbitmqctl list_users | grep otelu > /dev/null; then
+    echo "creating user: ${USER} . . ."
+    rabbitmqctl add_user ${USER} ${ROOT_PASS} > /dev/null
+
+    echo "Configuring ${USER} permissions. . ."
+    rabbitmqctl set_user_tags ${USER} administrator > /dev/null
+    rabbitmqctl set_permissions -p /  ${USER} ".*" ".*" ".*" > /dev/null
+fi
 
 echo "create exchange and queue. . ."
 rabbitmqadmin -u ${USER} -p ${ROOT_PASS} declare exchange --vhost=/ name=some_exchange type=direct > /dev/null

